dm/docs/RFCS/20220202_optimist_shard_mode_support_conflict_ddl.md

```diff
@@ -0,0 +1,155 @@
+# Proposal: DM Optimist Shard Mode Support Conflict DDL
+
+- Author(s):    [gmhdbjd](https://github.com/gmhdbjd)
+- Last updated: 2022-03-03
+
+## Background
+
+Current optimist shard mode does not support three types of DDLs.
+
+- DDLs that make a table "big and small", such as `alter table tb rename column a to b`, which in a sense is equivalent to deleting `a` and adding `b` to table `tb`, making the table "big and small".
+- DDLs that make the table incomparable, such as `alter table tb change a from int to varchar`, make it impossible to compare the size of `varchar` and `int`. For this kind of DDL, we can resolve it according to TiDB's fieldTypeMergeRules case by case, but we still need to consider a generic solution in order to be as compatible as possible with all DDLs. In a sense, this DDL also makes the table "big and small" (remove `a int`, add `a varchar`)
+- Adding NotNULL NoDefault columns, such as `alter table tb add column a not null`, which also makes the table incomparable (INT NOT NULL No Default vs Nil). In a sense, this DDL also makes the table "big and small" (remove nil, add `a int not null`)
```

Comment:
```suggestion
- Adding NotNULL NoDefault columns, such as `alter table tb add column a not null`, which also makes the table incomparable (INT NOT NULL No Default vs Nil). In a sense, this DDL also makes the table "bigger and smaller" in column level (remove nil, add `a int not null`)
```

Why can't we merge case 2 &3 to case 1? They are all makes the table "bigger and smaller". Does case 1 are compared at multiple column level?

Reply:
This is to illustrate that all three DDLs are bigger and smaller, which is why they are split up